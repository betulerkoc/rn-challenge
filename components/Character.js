import React, {useState, useEffect}  from 'react';
import {SafeAreaView, StyleSheet, Image, TouchableOpacity} from 'react-native';

const Character = ({url, navigation}) => {
  const [character, setCharacter] = useState(null);

  const fetchData = async () => {
    const response = await fetch(url);
    const json = await response.json();
    setCharacter({data: json});
  };

  useEffect(() => {
    fetchData();
  }, []);

  return (
    <SafeAreaView>
      {character && (
        <TouchableOpacity
          onPress={() =>
            navigation.navigate('CharacterDetail', {
              id: character.data.id,
            })
          }>
          <Image
            style={styles.tinyLogo}
            source={{
              uri: character.data.image,
            }}
          />
        </TouchableOpacity>
      )}
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  sectionContainer: {
    marginTop: 32,
    paddingHorizontal: 24,
  },
  tinyLogo: {
    width: 150,
    height: 150,
    margin: 10,
  },
});

export default Character;
