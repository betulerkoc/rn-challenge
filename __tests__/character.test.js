import 'react-native';
import React from 'react';
import Character from '../components/Character';
import renderer from 'react-test-renderer';

test('Character snapshot', () => {
    const snap = renderer.create(
        <Character/>
    ).toJSON();
    expect(snap).toMatchSnapshot();
}) 