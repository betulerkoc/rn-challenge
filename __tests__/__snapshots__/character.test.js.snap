// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
/>
`;
